src/DankCord/DankCord.py

```diff
@@ -45,94 +50,105 @@ def _get_commands(self, channel_id : int = None):
     for command_data in response.data.get('application_commands'):
       data[command_data["name"]] = command_data
     with open("commands.json", "w+") as f:
-      f.write(json.dumps(data, indent = 2, sort_keys = False))
+      f.write(json.dumps(data, indent=2, sort_keys=False))
     return response
 
-  def _get_command_info(self, name : str):
+  def _get_command_info(self, name: str):
     with open("commands.json", "r+") as f:
       data = json.loads(f.read())
     return data[name]
-  
-  def _confirm_command_ran(self, nonce_used : str, timeout : int):
+
+  def _confirm_command_ran(self, name: str, nonce_used: str, timeout: int):
     end = time.time() + timeout
-    while time.time() < end:
+    while True:
       recieved_nonce = json.loads(self.ws.recv())
-      if recieved_nonce["d"] == nonce_used:
-        return True
-    return False
+      try:
+        if recieved_nonce["d"]["interaction"]["user"]["id"] == str(self.user_id) and recieved_nonce["d"]["interaction"]["name"] == name and recieved_nonce["d"]["channel_id"] == str(self.channel_id):
+          return recieved_nonce["d"]
+        time.sleep(0.1)
+        if time.time() > end:
+          break
+      except:
+        pass
+    return None
 
   def _OptionsBuilder(self, name, type_, **kwargs):
     options = [
-      {
-        "type": type_,
-        "name": name,
-        "options": [
-        ]
-      }
+        {
+            "type": type_,
+            "name": name,
+            "options": [
+            ]
+        }
```

Comment:
Please make sure these are 2 space indents as well, not 4.

Reply:
Done